base/arrayshow.jl

```diff
@@ -416,7 +416,7 @@ _show_empty(io, X) = nothing # by default, we don't know this constructor
 function show(io::IO, X::AbstractArray)
     ndims(X) == 1 && return show_vector(io, X)
     prefix = typeinfo_prefix(io, X)
-    io = IOContext(io, :typeinfo => eltype(X), :compact => true)
+    io = IOContext(io, :typeinfo => eltype(X), :compact => get(io, :compact, true))
```

Comment:
Do you actually need to pass `:compact` manually? I thought `io` would inherit the property from its parent if it's set.

Reply:
But here we want to merely pass down the inherited `:compact` property only if it's set: if unset, we want to explicitely set it to `:compact=>true` when displaying the array elements.